include/boost/numeric/ublas/tensor/dynamic_strides.hpp

```diff
@@ -149,46 +156,60 @@ class basic_strides
 
     [[nodiscard]] inline
     constexpr const_reference back () const{
-        return _base[_base.size() - 1];
+        return _base.back();
     }
 
     [[nodiscard]] inline
     constexpr reference back (){
-        return _base[_base.size() - 1];
+        return _base.back();
     }
 
     [[nodiscard]] inline
-    constexpr bool empty() const{
+    constexpr bool empty() const noexcept{
         return _base.empty();
     }
 
     [[nodiscard]] inline
-    constexpr size_type size() const{
+    constexpr size_type size() const noexcept{
         return _base.size();
     }
 
     [[nodiscard]] inline
-    constexpr const_iterator begin() const{
+    constexpr const_iterator begin() const noexcept{
         return _base.begin();
     }
 
     [[nodiscard]] inline
-    constexpr const_iterator end() const{
+    constexpr const_iterator end() const noexcept{
         return _base.end();
     }
 
     inline
-    constexpr void clear() {
+    constexpr void clear() noexcept{
         this->_base.clear();
     }
 
     [[nodiscard]] inline
-    constexpr base_type const& base() const{
+    constexpr base_type const& base() const noexcept{
         return this->_base;
     }
+
+    [[nodiscard]] inline
+    constexpr const_reverse_iterator
+    rbegin() const noexcept
+    {
+        return _base.rbegin();
+    }
+
+    [[nodiscard]] inline
+    constexpr const_reverse_iterator
+    rend() const noexcept
+    {
+        return _base.rend();
+    }
     
-protected:
```

Comment:
I don't understand why were these `protected` members in the beginning. C++ ISO defers using protected members.

Reply:
The earlier version of tensor implementation had to use the base class data members to avoid friend class I had to use it.